adapters/klab.ogc/src/main/java/org/integratedmodelling/klab/stac/STACEncoder.java

```diff
@@ -122,22 +125,25 @@ public void getEncodedData(IResource resource, Map<String, String> urnParameters
                     (int) grid.getXCells(), (int) grid.getYCells());
 
             Integer srid = items.get(0).getEpsg();
-            CoordinateReferenceSystem outputCrs = CrsUtilities.getCrsFromSrid(srid);
+            CrsUtilities.getCrsFromSrid(srid);
             ReferencedEnvelope regionEnvelope = new ReferencedEnvelope(region.toEnvelope(),
-                    DefaultGeographicCRS.WGS84).transform(outputCrs, true);
+                    space.getProjection().getCoordinateReferenceSystem());
             RegionMap regionTransformed = RegionMap.fromEnvelopeAndGrid(regionEnvelope, (int) grid.getXCells(), (int) grid.getYCells());
             String assetId = resource.getParameters().get("asset", String.class);
-            
-            
-            // TODO Inigo check this. I think this needs some discussion. Allow transform 
-            // ensures the process to finish, but I would not bet on the resulting data.
-            boolean allowTransform = true;
-            HMRaster outRaster = HMStacCollection.readRasterBandOnRegion(regionTransformed, assetId, items, true, lpm);
+
+            Set<Integer> ESPGsAtItems = items.stream().map(i -> i.getEpsg()).collect(Collectors.toUnmodifiableSet());
+            if (ESPGsAtItems.size() > 1) {
+                scope.getMonitor().warn("Multiple ESPGs found on the items " + ESPGsAtItems.toString() + ". The transformation process could affect the data.");
+            }
+
+            // Allow transform ensures the process to finish, but I would not bet on the resulting data.
+            final boolean allowTransform = true;
+            HMRaster outRaster = HMStacCollection.readRasterBandOnRegion(regionTransformed, assetId, items, allowTransform, lpm);
 
             coverage = outRaster.buildCoverage();
```

Comment:
Since the epsg of the raster is unknown and probably != from the one of the context, will you need to do a reprojection here?

Reply:
Would the `allowTransform` affect observations where the ESPG is always the same? I think that, while the reprojection is not ideal, on some cases it is the only way we have for the process to finish.